src/Windows/Avalonia.Win32/Win32Platform.cs

```diff
@@ -129,21 +129,34 @@ public bool HasMessages()
 
         public void ProcessMessage()
         {
-            UnmanagedMethods.MSG msg;
-            UnmanagedMethods.GetMessage(out msg, IntPtr.Zero, 0, 0);
-            UnmanagedMethods.TranslateMessage(ref msg);
-            UnmanagedMethods.DispatchMessage(ref msg);
+
+            if (UnmanagedMethods.GetMessage(out var msg, IntPtr.Zero, 0, 0) > 0)
+            {
+                UnmanagedMethods.TranslateMessage(ref msg);
+                UnmanagedMethods.DispatchMessage(ref msg);
+            }
+            else
```

Comment:
In this case there will never be zero?

It's just an error [only when -1](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmessagew#return-value)